pkg/server/server_integration_test.go

```diff
@@ -91,14 +126,26 @@ func readMessage(t *testing.T, ws *websocket.Conn) messaging.Message {
 	return m
 }
 
-func assertSameMessages(t *testing.T, from string, sent agent.ClientMessage, recv messaging.Message) {
+func assertSameClientMessages(t *testing.T, from string, sent agent.ClientMessage, recv messaging.Message) {
 	if from != recv.From {
 		t.Errorf("received message sent from %q, but wanted from %s", recv.From, from)
 	}
 	if sent.To != recv.To {
 		t.Errorf("received message sent to %q, but wanted to %s", recv.To, sent.To)
 	}
 	if !reflect.DeepEqual(sent.Payload, recv.Payload) {
-		t.Errorf("received message body is %v, but wanted to %v", recv.Payload, sent.Payload)
+		t.Errorf("received message body is %v, but wanted %v", recv.Payload, sent.Payload)
+	}
+}
+
+func assertSameMessages(t *testing.T, sent messaging.Message, recv messaging.Message) {
```

Comment:
If you move it to the agent then I think you also won't need this method which is almost same as the above and looks like duplicated code. Then you can just assert on the type and peer in the message that are recorded on the mocked broker.